node/querystring.ts

```diff
@@ -371,38 +372,42 @@ export function stringify(
   const encode = options ? options.encodeURIComponent : escape;
   const convert = options ? encodeStringifiedCustom : encodeStringified;
 
-  if (obj !== null && typeof obj === "object") {
-    const keys = Object.keys(obj);
-    const len = keys.length;
-    let fields = "";
-    for (let i = 0; i < len; ++i) {
-      const k = keys[i];
-      const v = obj[k];
-      let ks = convert(k, encode);
-      ks += eq;
-
-      if (Array.isArray(v)) {
-        const vlen = v.length;
-        if (vlen === 0) continue;
-        if (fields) {
-          fields += sep;
-        }
-        for (let j = 0; j < vlen; ++j) {
-          if (j) {
+  try {
+    if (obj !== null && typeof obj === "object") {
+      const keys = Object.keys(obj);
+      const len = keys.length;
+      let fields = "";
+      for (let i = 0; i < len; ++i) {
+        const k = keys[i];
+        const v = obj[k];
+        let ks = convert(k, encode);
+        ks += eq;
+
+        if (Array.isArray(v)) {
+          const vlen = v.length;
+          if (vlen === 0) continue;
+          if (fields) {
+            fields += sep;
+          }
+          for (let j = 0; j < vlen; ++j) {
+            if (j) {
+              fields += sep;
+            }
+            fields += ks;
+            fields += convert(v[j], encode);
+          }
+        } else {
+          if (fields) {
             fields += sep;
           }
           fields += ks;
-          fields += convert(v[j], encode);
-        }
-      } else {
-        if (fields) {
-          fields += sep;
+          fields += convert(v, encode);
         }
-        fields += ks;
-        fields += convert(v, encode);
       }
+      return fields;
     }
-    return fields;
+  } catch (_) {
+    throw new ERR_INVALID_URI();
```

Comment:
Is it possible to narrow the kinds of error here? Do you know what is the exact error here when Invalid surrogate pair is given?

Reply:
I needed to use `encodeStr` for exact error handling when an invalid surrogate pair is specified.
https://github.com/nodejs/node/blob/229a182823aa5e60ee5420170ec11e9a541f2163/lib/internal/querystring.js#L90-L97

and I made sure not to export `encodeStr` as it should only be used internally. https://github.com/denoland/deno_std/pull/1505/commits/35eec80f49e7f0f1769f22809fd13262e54f25c3